File: Rust/fibonacci.rs
struct Fibonacci {
    curr: u64,
    prev: u64,
}

impl Iterator for Fibonacci {
    type Item = u64;
    
    fn next(&mut self) -> Option<u64> {
        let temp = self.curr;
        self.curr = self.curr + self.prev;
        self.prev = temp;
        
        Some(temp)
    }
}

fn fibonacci() -> Fibonacci {
    Fibonacci { curr: 1, prev: 1 }
}

fn main() {
    println!("{}", fibonacci()
             .take_while(|&x| x < 4000000)
             .filter(|x| x % 2 == 0)
             .fold(0, |a,b| a + b));
}

File: javascript/fibonacci.js
function fibonacci (number) {
    var varFibonacci = [0, 1];
  
    if (number <= 2) return 1;
    
    for (var i = 2; i <= number; i++) {
        varFibonacci[i] = varFibonacci[i-1] + varFibonacci[i-2];
    }
  
    return varFibonacci[number];
}